compaction/compaction_manager.cc

```diff
@@ -539,11 +539,10 @@ future<> compaction_manager::stop_tasks(std::vector<lw_shared_ptr<task>> tasks,
 }
 
 future<> compaction_manager::stop_ongoing_compactions(sstring reason) {
-    cmlog.info("Stopping {} ongoing compactions due to {}", get_compactions().size(), reason);
-
     // Wait for each task handler to stop. Copy list because task remove itself
     // from the list when done.
     auto tasks = boost::copy_range<std::vector<lw_shared_ptr<task>>>(_tasks);
+    cmlog.info("Stopping {} ongoing compactions due to {}", tasks.size(), reason);
```

Comment:
This will report tasks that aren't running. So they aren't ongoing compactions.

Reply:
1. They are going to be stopped so we better report what we do (and nt say we're stopping 0 ongoing compactions and then go and stop 17 tasks), so maybe we should change the message to say "Stopping {} compaction tasks..."?
2. If they aren't running, they shouldn't be listen in _tasks
3. This is consistent with `stop_ongoing_compactions(sstring reason, column_family* cf)` below.

Reply:
Why does the user case about tasks? They're invisible if they aren't running.

Reply:
compaction tasks that are not `compaction_running` (therefore not currently counted as "ongoing" here)
may either be:
- postponed (due to compaction throttling considerations)
- or sleeping before retry.

in both cases `stop_ongoing_compactions` is expected to stop these tasks.

From the user standpoint all this happens under the hood so it doesn't matter for the user if the tasks are active or postponed or waiting for retry, whatever...  I think that the log message is intended mostly for developers / support, not the end user.

Reply:
we need a reword at least, non running tasks aren't ongoing compactions. so it will confuse both us and users. 

ex: Stopping {} ongoing compactions (out of {} tasks) due to {}

Reply:
Done in v2